src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -547,6 +547,8 @@ class ShenandoahHeap : public CollectedHeap {
 
   bool is_in(const void* p) const;
 
+  bool is_in_young(const void* p) const;
```

Comment:
This overrides CollectedHeap method, right? So either put 'virtual' here to indicate this, or even better 'override'. I believe we can use that new C++11 specifier now...

Reply:
This method does not exist in CollectedHeap. It does exist in GenCollectedHeap, but ShenandoahHeap is not a subclass of the latter. (GenCollectedHeap is rather specialized, not suitable for us. It assumes contiguous generation spaces. G1 does not use it either.)